lib/src/event_manager.dart

```diff
@@ -0,0 +1,245 @@
+import 'dart:async';
+
+import 'package:nyxx/nyxx.dart';
+import 'package:nyxx_interactions/nyxx_interactions.dart';
+
+import 'commands.dart';
+import 'commands/chat_command.dart';
+import 'commands/message_command.dart';
+import 'commands/user_command.dart';
+import 'context/autocomplete_context.dart';
+import 'context/base.dart';
+import 'context/chat_context.dart';
+import 'context/component_context.dart';
+import 'errors.dart';
+import 'mirror_utils/mirror_utils.dart';
+import 'util/util.dart';
+import 'util/view.dart';
+
+/// A handler for events incoming from nyxx and nyxx_interactions, and listeners associated with
+/// those events.
+///
+/// This class will listen to events from nyxx and nyxx_interactions, transform them into a
+/// nyxx_commands class using [CommandsPlugin.contextManager] if needed and emit them to the correct
+/// command handler or listener.
+class EventManager {
```

Comment:
can you provide some tests for this class? There is a lot going on and would be good to have it tested